Modules/_functoolsmodule.c

```diff
@@ -932,15 +932,31 @@ _functools_cmp_to_key_impl(PyObject *module, PyObject *mycmp)
 
 /* reduce (used to be a builtin) ********************************************/
 
-// Not converted to argument clinic, because of `args` in-place modification.
-// AC will affect performance.
+/*[clinic input]
+_functools.reduce
+
+    function as func: object
+    iterable as seq: object
+    initial as result: object(c_default="NULL") = _functools._initial_missing
```

Comment:
Why not simply use NULL?
```suggestion
    initial as result: object = NULL
```